### ISSUE TITLE
jqueryui not found

### ISSUE THREAD

**Issue by svenXY** [cosminadrianpopescu/muttcube#1]
Hi,

you require jqueryui, but I cannot find the plugin.

**Comment by cosminadrianpopescu** [comment]
The jqueryui comes as a plugin of roundcube. The default theme is relating on it. If you have a theme which does not have jquery ui anyway, this plugin will probably not work. Try and activate the jqueryui plugin in roundcube and see if this solves the problem.

**Comment by svenXY** [comment]
Hi,

I enabled it as a plugin in config.inc.php, but the error message remains the same:
```Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for cosminadrianpopescu/muttcube dev-master -> satisfiable by cosminadrianpopescu/muttcube[dev-master].
    - cosminadrianpopescu/muttcube dev-master requires jqueryui * -> no matching package found.
```

**Comment by cosminadrianpopescu** [comment]
what version of roundcube are you using?

**Comment by svenXY** [comment]
I had been using the one from debian when I first found this problem, but have since migrated to git master (1.2-git [GIT 20160425.0912]), wehre the problem persists.

To me it looks like composer wants to install a plugin that is already installed, but not through composer.

**Comment by cosminadrianpopescu** [comment]
And what skin are you using? Because as i was saying, if you are using larry skin, this should have already the jquery-ui included since it's depending on it. But if you are using other skin, I am not sure the plugin will work anyway (I mean muttcube). However, if you tell me what skin you are using and where to get it, I can try and see what is the issue.

**Comment by svenXY** [comment]
I am using the larry skin. From the About-screen, I can also see that jqueryui is loaded properly.

My question would be: how does composer.phar detect if a non-composer-added plugin (like e.g. jqueryui) is present? That means - would your plugin install and work if jqueryui is enabled but removed from your dependencies?

**Comment by cosminadrianpopescu** [comment]
yes, it should work.

**Comment by svenXY** [comment]
OK, then how can I adjust this? Can I interfere with composer and remove (or ignore) the dependency?

**Comment by cosminadrianpopescu** [comment]
From the composer.json, just remove the line: 

  "require": {
        "jqueryui": "*"
    },

**Comment by svenXY** [comment]
So I 

- clone this into some directory (or a certain one???), 
- then adjust the composer.json 
- and run php composer.phar install?

**Comment by svenXY** [comment]
I cloned into ../plugins/muttcube and ran composer from there.
OK, that seems to have worked, but did not add the plugin to config.inc.php.
Anyway, it works now, thanks for your help.

One last question: / and esc do not seem to work on my linux/firefox combination.
/ starts the firefox search - I guess I can live with that, but ESC just does nothing.

Any ideas?

**Comment by cosminadrianpopescu** [comment]
It depends on what other plugins you have on firefox. You might have a vimperator (especially if / starts the search). If this is the case, disable the vimperator for this site. I am using qutebrowser, which is a vimperator like browser and I disable the vim keybinding for the roundcube sites. 

On clean browsers (without plugins) the shortcut for search should be "Ctrl + F". And it should work without any issues.

**Comment by svenXY** [comment]
Nope, no vimperator. / (and ') are core firefox keyboard shortcuts and I cannot disable them easily.

But I found this:
http://stackoverflow.com/a/7894325

which would allow you to disable it from jquery

**Comment by cosminadrianpopescu** [comment]
The thing is that I think you need to put that on keyup and I am listening to keydown event. But I've just tested in a clean firefox (no extensions) and I don't have a find for the `/` shortcut. It works without problems. Can you try to create a new profile in your firefox (with no extension) and try again?

Also, in chrome there are no issues.

**Comment by svenXY** [comment]
Am 29.4.2016 16:20, schrieb Cosmin Popescu:
> The thing is that I think you need to put that on keyup and I am
> listening to keydown event. But I've just tested in a clean firefox
> (no extensions) and I don't have a find for the / shortcut. It works
> without problems. Can you try to create a new profile in your firefox
> (with no extension) and try again?

I did and the issue persists.


-- 

Sven Hergenhahn

heidelberg.germany.europe.earth

**Comment by svenXY** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=942269 - they are hard-coded and cannot be remapped - at least that's what this seems to indicate.

**Comment by svenXY** [comment]
and again - this http://www.unknownerror.org/opensource/schneiderandre/popping/q/stackoverflow/7893972/how-do-i-disable-the-slash-and-apostrophe-keys-from-popping-up-quick-find-in-fir might help to fix it from your side.

**Comment by cosminadrianpopescu** [comment]
The thing is that the muttcube plugin should even remap the `Ctrl+F`. This is why I am saying that it's strange. Because I hooked the `keydown` event and I am preventing the default. It works in Chrome and Firefox. Did you try on a clean profile without any extensions? Because I suspect that it's an extension that you have installed. 

Also, what are `j` and `k` shortcuts doing? They should go to the next and previous message in the messages list? 

I am thinking that maybe you have other roundcube plugins for the shortcuts.

**Comment by svenXY** [comment]
Hi,

Am 3.5.2016 12:44, schrieb Cosmin Popescu:
> Did you try on a clean profile without any extensions?
> Because I suspect that it's an extension that you have installed.

Nope. As said before, I already tried with a new empty profile.

> Also, what are j and k shortcuts doing? They should go to the next and
> previous message in the messages list?

j, k and so on work as expected.
Only / does not.

> I am thinking that maybe you have other roundcube plugins for the
> shortcuts.

I had another shortcut plugin installed, but I disabled it before 
starting to use muttcube.


-- 
Sven Hergenhahn

heidelberg.germany.europe.earth

**Comment by cosminadrianpopescu** [comment]
Even `Ctrl+F` and `Ctrl+B` work as expected? They are supposed to do a page up and page down

**Comment by svenXY** [comment]
Am 4.5.2016 09:13, schrieb Cosmin Popescu:
> Even Ctrl+F and Ctrl+B work as expected? They are supposed to do a
> page up and page down

Yes, that works as expected.


-- 

Sven Hergenhahn

heidelberg.germany.europe.earth

**Comment by cosminadrianpopescu** [comment]
Did you also tested in Chrome? Because as you can see even `Ctrl + B` (which is the normal shortcut for bookings) and `Ctrl + F` are overriten. So, I really don't understand how `/` cannot be overriden (this is not even a classical shortcut).

**Comment by svenXY** [comment]
old and most probably no progress to be expected - closing
